core/deployment/src/main/java/io/quarkus/deployment/pkg/steps/JarResultBuildStep.java

```diff
@@ -593,23 +616,54 @@ public NativeImageSourceJarBuildItem buildNativeImageJar(CurateOutcomeBuildItem
             List<GeneratedClassBuildItem> generatedClasses,
             List<GeneratedNativeImageClassBuildItem> nativeImageResources,
             List<GeneratedResourceBuildItem> generatedResources,
-            MainClassBuildItem mainClassBuildItem) throws Exception {
-        Path thinJarDirectory = outputTargetBuildItem.getOutputDirectory()
+            MainClassBuildItem mainClassBuildItem,
+            List<UberJarRequiredBuildItem> uberJarRequired) throws Exception {
+        Path targetDirectory = outputTargetBuildItem.getOutputDirectory()
                 .resolve(outputTargetBuildItem.getBaseName() + "-native-image-source-jar");
-        IoUtils.recursiveDelete(thinJarDirectory);
-        Files.createDirectories(thinJarDirectory);
-        copyJsonConfigFiles(applicationArchivesBuildItem, thinJarDirectory);
-
-        Path runnerJar = thinJarDirectory
-                .resolve(outputTargetBuildItem.getBaseName() + packageConfig.runnerSuffix + ".jar");
-        Path libDir = thinJarDirectory.resolve(LIB);
-        Files.createDirectories(libDir);
+        IoUtils.recursiveDelete(targetDirectory);
+        Files.createDirectories(targetDirectory);
 
         List<GeneratedClassBuildItem> allClasses = new ArrayList<>(generatedClasses);
         allClasses.addAll(nativeImageResources.stream()
                 .map((s) -> new GeneratedClassBuildItem(true, s.getName(), s.getClassData()))
                 .collect(Collectors.toList()));
 
+        boolean uberJarRequested = !uberJarRequired.isEmpty() || packageConfig.uberJar;
```

Comment:
I think this should only be for windows. We just want one format to build the native image from in general, this is an intermediate state so it should not matter to the user, but having two different formats means more possible things to debug.

Reply:
@stuartwdouglas Can you check if my last change is what you had in mind?